core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -52,13 +51,23 @@ import org.apache.spark.util.{AccumulatorV2, ThreadUtils, Utils}
  * acquire a lock on us, so we need to make sure that we don't try to lock the backend while
  * we are holding a lock on ourselves.
  */
-private[spark] class TaskSchedulerImpl(
+private[spark] class TaskSchedulerImpl private[scheduler](
```

Comment:
just to make sure I understand -- do you want to eliminate having it be a parameter for the constructor to `TaskSchedulerImpl` completely, so that it is always created from the conf?  I think that would prevent proper testing, since I couldn't eg. pass in a mock.

Or keep the constructor, but just make `blacklistTracker` completely private, (not `private[scheduler]`)?  That should be straightforward, all the tests which directly create a TaskSetManager would just use `None` for the blacklist.